File: binary_search/binary_search.go
package main

import "fmt"

func binarySearch(array []int, num int) int {

	var low, high int
	low = array[0]
	high = array[len(array)-1]

	if num > high || num < low {
		return -1
	}

	for low <= high {
		mid := low + (high-low)/2
		switch {
		case num < array[mid]:
			high = mid - 1

		case num > array[mid]:
			low = mid + 1

		case num == array[mid]:
			return mid
		}
	}
	return -1
}

func main() {

	fmt.Println("Binary search:")
	array := []int{0, 2, 4, 6, 8, 10, 12, 14, 16, 18}
	index := binarySearch(array, 10)
	if index == -1 {
		fmt.Println("Number not found")
	} else {
		fmt.Println("Index: ", index)
		fmt.Println("array[", index, "] = ", array[index])
	}
}

File: binary_search/c/binary_search.java
public class Main {
   int binarySearch(int arr[], int l, int r, int x) {
      if (r >= l) {
           int mid = l + (r - l)/2;

           if (arr[mid] == x)  return mid;

           if (arr[mid] > x) return binarySearch(arr, l, mid-1, x);

           return binarySearch(arr, mid+1, r, x);
      }
      return -1;
   }

   public static void main(String[] args){
      
      int[] arr = {1,2,3,4,5};
      int valueFind = 2;
      System.out.println(binarySearch(arr,0,arr.length,valueFind));
      
   }
}
